packages/eas-cli/src/commands/fingerprint/generate.ts

```diff
@@ -74,12 +96,35 @@ export default class FingerprintGenerate extends EasCommand {
       platform = await selectRequestedPlatformAsync();
     }
 
+    let env: Env | undefined = undefined;
     if (environment) {
       Log.log(`🔧 Using environment: ${environment}`);
+      env = { ...(await getServerSideEnvironmentVariablesAsync()), EXPO_NO_DOTENV: '1' };
+    } else if (buildProfileName) {
+      Log.log(`🔧 Using build profile: ${buildProfileName}`);
+      const easJsonAccessor = EasJsonAccessor.fromProjectPath(projectDir);
+      const buildProfile = (
+        await getProfilesAsync({
+          type: 'build',
+          easJsonAccessor,
+          platforms: [appPlatformToPlatform(platform)],
+          profileName: buildProfileName ?? undefined,
+          projectDir,
+        })
+      )[0];
+      if (!buildProfile) {
+        throw new Error(`Build profile ${buildProfile} not found for platform: ${platform}`);
+      }
+      const configResult = await evaluateConfigWithEnvVarsAsync({
+        buildProfile: buildProfile.profile,
+        buildProfileName: buildProfile.profileName,
+        graphqlClient,
+        getProjectConfig: getDynamicPrivateProjectConfigAsync,
+        opts: { env: buildProfile.profile.env },
+      });
+      env = configResult.env;
```

Comment:
I've skipped adding `EXPO_NO_DOTENV: '1'` in this case to keep it consistent with `eas build`. Additional reasoning:
- `eas.json` may not define server side variables, in which case we'd want to use `.env` to get env vars
- will leave it to the user to define `EXPO_NO_DOTENV` in `eas.json` as it is not clear what type of env var system they are using.